java/pgv-java-stub/src/main/java/io/envoyproxy/pgv/StringValidation.java

```diff
@@ -153,12 +153,32 @@ public static void uriRef(String field, String value) throws ValidationException
         }
     }
 
-    private static final Pattern uuidPattern = Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
-    public static void uuid(String field, String value) throws ValidationException {
-        Matcher matcher = uuidPattern.matcher(value);
-        if (!matcher.matches()) {
-            throw new ValidationException(field, enquote(value), "should be a valid uuid");
+    public static void uuid(final String field, final String value) throws ValidationException {
```

Comment:
Since finite state machines are harder to understand from code than regexes, can you please add a doc comment explaining that this method matches rfc4122 string serialized UUIDs?